From e71b1b9ced8b2478446d931a47a94296b72b833b Mon Sep 17 00:00:00 2001
From: Terry Lv <r65388@freescale.com>
Date: Fri, 17 Sep 2010 13:25:29 +0800
Subject: [PATCH] ENGR00131662: Fix that iram boot don't work

iram boot don't work.
The reason is that ivt plugins copy too much data
than that iram can hold.

Signed-off-by: Terry Lv <r65388@freescale.com>
---
 board/freescale/mx50_arm2/flash_header.S |   10 +++++-----
 include/configs/mx50_arm2_iram.h         |    2 +-
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/board/freescale/mx50_arm2/flash_header.S b/board/freescale/mx50_arm2/flash_header.S
index 8b0edca..94549bb 100644
--- a/board/freescale/mx50_arm2/flash_header.S
+++ b/board/freescale/mx50_arm2/flash_header.S
@@ -39,7 +39,7 @@ self_ptr:          .long 0xF8006400
 app_code_csf:      .long 0x0 	       /* reserve 4K for csf */
 reserv2:           .long 0x0
 boot_data:         .long 0xF8006000
-image_len:         .long 8*1024        /* Can copy upto 72K, OCRAM free space */
+image_len:         .long 4*1024        /* Can copy upto 72K, OCRAM free space */
 plugin:            .long 0x1  	       /* Enable plugin flag */
 
 /* Second IVT to give entry point into the bootloader copied to DDR */
@@ -1166,10 +1166,10 @@ after_calling_rom___pu_irom_hwcnfg_setup:
 	mov r0, #1
 	bx lr          /* return back to ROM code */
 
-DDR_DEST_ADDR:    .word   0x77800000
-COPY_SIZE:        .word   0x40000
-BOOT_DATA:        .word   0x77800000
-                  .word   0x40000 /*data be copied by pu_irom_hwcnfg_setup()*/
+DDR_DEST_ADDR:    .word   TEXT_BASE
+COPY_SIZE:        .word   _end - TEXT_BASE
+BOOT_DATA:        .word   TEXT_BASE
+                  .word   _end - TEXT_BASE /*data be copied by pu_irom_hwcnfg_setup()*/
                   .word   0
 
 #endif
diff --git a/include/configs/mx50_arm2_iram.h b/include/configs/mx50_arm2_iram.h
index 0e7f4b4..a0a5e99 100644
--- a/include/configs/mx50_arm2_iram.h
+++ b/include/configs/mx50_arm2_iram.h
@@ -41,7 +41,7 @@
 */
 
 #define CONFIG_MX50_HCLK_FREQ	24000000
-#define CONFIG_SYS_PLL2_FREQ    600
+#define CONFIG_SYS_PLL2_FREQ    400
 #define CONFIG_SYS_AHB_PODF     4
 #define CONFIG_SYS_AXIA_PODF    1
 #define CONFIG_SYS_AXIB_PODF    2
-- 
1.5.4.4

